lib/avromatic/model/configurable.rb

```diff
@@ -8,6 +8,17 @@ module Model
     module Configurable
       extend ActiveSupport::Concern
 
+      # Wraps a reference to a field so we can access both the string and symbolized versions of the name
+      # without repeated memory allocations.
+      class FieldReference
+        attr_reader :name, :name_sym
+
+        def initialize(name)
+          @name = -name
```

Comment:
Should we freeze this string after dup'ing?

Reply:
It's a bit cryptic but the urinary `-` operator does return a frozen string: https://ruby-doc.org/core-2.7.2/String.html#method-i-2D-40